packages/react-native-audio-api/common/cpp/audioapi/utils/Result.hpp

```diff
@@ -0,0 +1,286 @@
+  }
+
+  /// @brief Returns the contained Ok value.
+  /// @throws std::runtime_error if the value is an Err.
+  [[nodiscard]] const T& unwrap() const & {
+    if (!is_ok_) {
+      throw std::runtime_error("Called unwrap on an Err value");
+    }
+    return ok_value;
+  }
+
+  /// @brief Returns the contained Ok value or a default. Consumes self.
+  [[nodiscard]] T unwrap_or(T&& default_value) && {
+    if (is_ok_) {
+      return std::move(ok_value);
+    }
+    return std::move(default_value);
+  }
+
+  /// @brief Returns the contained Ok value or computes it from a closure. Consumes self.
+  [[nodiscard]] T unwrap_or_else(const std::function<T(E&&)>& func) && {
+    if (is_ok_) {
+      return std::move(ok_value);
+    }
+    return func(std::move(err_value));
+  }
+
+  /// @brief Returns the contained Ok value without checking. UB if Err.
+  [[nodiscard]] T unwrap_unchecked() && noexcept {
+    return std::move(ok_value);
+  }
+
+  /// @brief Returns the contained Ok value without checking. UB if Err.
+  [[nodiscard]] const T& unwrap_unchecked() const & noexcept {
+    return ok_value;
+  }
+
+  /// @brief Returns the contained Err value, consuming the Result.
+  /// @throws std::runtime_error if the value is Ok.
+  [[nodiscard]] E unwrap_err() && {
+    if (is_ok_) {
+      throw std::runtime_error("Called unwrap_err on an Ok value");
+    }
+    return std::move(err_value);
+  }
+
+  /// @brief Returns the contained Err value.
+  /// @throws std::runtime_error if the value is Ok.
+  [[nodiscard]] const E& unwrap_err() const & {
+    if (is_ok_) {
+      throw std::runtime_error("Called unwrap_err on an Ok value");
+    }
+    return err_value;
+  }
+
+  /// @brief Returns the contained Err value, consuming the Result.
+  /// @throws std::runtime_error if the value is Ok.
+  [[nodiscard]] E expect_err(const std::string& msg) && {
+    if (is_ok_) {
+      throw std::runtime_error(msg);
+    }
+    return std::move(err_value);
+  }
+
+  /// @brief Returns the contained Err value.
+  /// @throws std::runtime_error if the value is Ok.
+  [[nodiscard]] const E& expect_err(const std::string& msg) const & {
+    if (is_ok_) {
+      throw std::runtime_error(msg);
+    }
+    return err_value;
+  }
+
+  /// @brief Returns the contained Err value without checking. UB if Ok.
+  [[nodiscard]] E unwrap_err_unchecked() && noexcept {
+    return std::move(err_value);
+  }
+
+  /// @brief Returns the contained Err value without checking. UB if Ok.
+  [[nodiscard]] const E& unwrap_err_unchecked() const & noexcept {
+    return err_value;
+  }
+
+  /// @brief Maps a Result<T, E> to Result<U, E> by applying a function to a contained Ok value.
+  template<typename TNew>
+  [[nodiscard]] std::enable_if_t<!std::is_void_v<T>, Result<TNew, E>> map(std::function<TNew(T&&)> ok_func) && {
+    if (is_ok_) {
+      return Result<TNew, E>::Ok(ok_func(std::move(ok_value)));
+    } else {
+      return Result<TNew, E>::Err(std::move(err_value));
+    }
+  }
```

Comment:
```suggestion
template <typename Type>
using isNotVoid = std::enable_if_t<!std::is_void_v<Type>>

  template<typename TNew, typename = isNotVoid<TNew>>
  [[nodiscard]] Result<TNew, E> map(std::function<TNew(T&&)> ok_func) && {
    if (is_ok_) {
      return Result<TNew, E>::Ok(ok_func(std::move(ok_value)));
    } else {
      return Result<TNew, E>::Err(std::move(err_value));
    }
  }
```
and we can utilize `isNotVoid` and this template syntax in all following cases for more readability